HW4_Uzun/AAmigo.py

```diff
@@ -0,0 +1,181 @@
+     Amino acids in the string should be indicated as one-letter symbols.
+
+    """
+    aa_seq = list(seq.upper())
+    aa_biochemistry = dict(
+        {'hydrophobic': ['G', 'A', 'V', 'L', 'I', 'P', 'F', 'M', 'W'], 'polar': ['S', 'T', 'C', 'N', 'Q', 'Y'],
+         '- charged': ['E', 'D'], '+ charged': ['K', 'H', 'R']})
+    profile = dict({'hydrophobic': 0.0, 'polar': 0.0, '- charged': 0.0, '+ charged': 0.0})
+
+    for amino_acid in aa_seq:
+        for group_name, group_list in aa_biochemistry.items():
+            if amino_acid in group_list:
+                profile[group_name] += 1
+
+    for group, count in profile.items():
+        profile[group] = round((count/len(seq)), 2)
+    return profile
+
+
+def amino_acid_substring(seq: str):
+    """
+
+    Searches for a substring of amino acids in the entire amino acid sequence.
+    Takes a string of amino acids and a substring, which should be found.
+    Returns the position where the searched one was found for the first time.
+    Amino acids in the string should be indicated as one-letter symbols.
+
+    """
+    aa_seq = list(seq)
+    aa_seq_upper = []
+    for sequences in aa_seq:
+        upper_case = sequences.upper()
+        aa_seq_upper.append(upper_case)
+    amino_acids = aa_seq_upper[:-1]
+    substring = aa_seq_upper[-1]
+    results = []
+    for sequences in amino_acids:
+        subst = sequences.find(substring)
+        results.append(subst)
+    return results
+
+
+def amino_acid_count(seq: str):
+    """
+
+    Finds how many times a particular sequence(s) occurs in the original one.
+    Takes a string of amino acids and a substring, which should be counted.
+    Returns the count of searched amino acids.
+    Amino acids in the string should be indicated as one-letter symbols.
+
+    """
+    aa_seq = list(seq)
+    aa_seq_upper = []
+    for sequences in aa_seq:
+        upper_case = sequences.upper()
+        aa_seq_upper.append(upper_case)
+    amino_acids = aa_seq_upper[:-1]
+    substring = aa_seq_upper[-1]
+    results = []
+    for sequences in amino_acids:
+        aa_count = sequences.count(substring)
+        results.append(aa_count)
+    return results
+
+
+def protein_length(*seq: str):
+    """
+
+    Calculate the length (number of amino acids) of a protein.
+    Takes a string of amino acids, returns the number.
+    Amino acids in the string should be indicated as one-letter symbols.
+
+    """
+    lengths = []
+
+    for sequences in seq:
+        lengths.append(len(sequences))
+
+    return lengths
+
+
+def essential_amino_acids(*seq: str):
+    """
+
+    Calculate the number of essential amino acids based on its amino acids sequence.
+    Takes a string of amino acids, returns only the essential amino acids.
+    Amino acids in the string should be indicated as one-letter symbols.
+
+    """
+    eaa_dictionary = ['H', 'I', 'K', 'L', 'M', 'F', 'T', 'W', 'V', 'h', 'i', 'k', 'l', 'm', 'f', 't', 'w', 'v']

```

Comment:
В смысле `eaa_dictionary`? Это же список:)))?
![image](https://github.com/Python-BI-2023/HW4_Functions2/assets/76266535/7a2a7546-d514-441b-ad36-7681940f2ca4)


Ну и `eaa` без контекста совсем не понятно что такое. Плюс указывать тип данных в названии переменной не всегда обязательно
```suggestion
    essential_amino_acids = ['H', 'I', 'K', 'L', 'M', 'F', 'T', 'W', 'V', 'h', 'i', 'k', 'l', 'm', 'f', 't', 'w', 'v']
```

И в конце концов это дело мы убираем в начало кода в константы и назваем целиком капсом, так вообще красота будет